### PR TITLE
[opendis6] new recipe

### PR DESCRIPTION
**Pull request by crhowell3**
Specify library name and version:  **opendis6/0.1.0**

I am the author of this library. It is a modern C++ implementation of IEEE 1278.1a-1998 (DIS v6).

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@crhowell3 Thank you for your contribution, please, consider the PR https://github.com/crhowell3/conan-center-index/pull/1 as my review and improvement to this PR.

File: recipes/opendis6/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
```
Comment (uilianries) [inline recipes/opendis6/all/conanfile.py:66]:
```suggestion

    def build_requirements(self):
        tool_requires("cmake/[>=3.22 <4]")

```

Well, we will need to install cmake too

Comment (crhowell3) [reply recipes/opendis6/all/conanfile.py:66]: Right, I will add that in.

**Comment by crhowell3** [comment]
Looks like Apple Clang doesn't know what the `[[nodiscard]]` attribute is... the minimum C++ standard is set to 17 in my conanfile, so not sure why that would be happening

**Comment by crhowell3** [comment]
This should be good for merge 👍🏻

File: recipes/opendis6/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        copy(self, pattern="LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "res"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.components["OpenDIS6"].libs = ["OpenDIS6"]
```
Comment (uilianries) [inline recipes/opendis6/all/conanfile.py:92]:
```suggestion
        self.cpp_info.libs = ["OpenDIS6"]
```

Components could be skipped here.

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/opendis6/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            "Visual Studio": "15",
+            "msvc": "191",
+            "gcc": "8.5",
+            "clang": "6",
+            "apple-clang": "14",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
```
Comment (RubenRBS) [inline recipes/opendis6/all/conanfile.py:46]:
```suggestion
```

File: recipes/opendis6/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, export_conandata_patches, get, rmdir
```
Comment (RubenRBS) [inline recipes/opendis6/all/conanfile.py:5]:
```suggestion
from conan.tools.files import copy, get, rmdir
```

File: recipes/opendis6/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.15)
+project(test_package LANGUAGES CXX)
+
+find_package(OpenDIS CONFIG REQUIRED)
+
+add_executable(test_package_dis test_package.cpp)
+target_link_libraries(test_package_dis PRIVATE OpenDIS::OpenDIS6)
```
Comment (RubenRBS) [inline recipes/opendis6/all/test_package/CMakeLists.txt:7]:
```suggestion
target_link_libraries(test_package_dis PRIVATE OpenDIS::OpenDIS6)
set_target_properties(test_package_dis PROPERTIES CXX_STANDARD 17)
```

We usually like to ensure the test package uses the cxx standard defined by the recipe :)

**Review by RubenRBS — APPROVED** [review]

File: recipes/opendis6/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "res"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["OpenDIS6"]
+        self.cpp_info.set_property("cmake_file_name", "OpenDIS")
+        self.cpp_info.set_property("cmake_target_name", "OpenDIS::OpenDIS6")
+        self.cpp_info.set_property("cmake_target_aliases", ["OpenDIS::DIS6","OpenDIS6"])
```
Comment (RubenRBS) [inline recipes/opendis6/all/conanfile.py:92]:
```suggestion
        self.cpp_info.set_property("cmake_target_aliases", ["OpenDIS::DIS6","OpenDIS6"])
        
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("m")
```

**Review by RubenRBS — APPROVED** [review]
Thanks a lot for your contribution, we appreciate it, I've pushed the last final changes to get this merged :)

**Review by jwillikers — APPROVED** [review]

**Pull request merged**